src/Product/ProductMetaHandler.php

```diff
@@ -143,10 +145,13 @@ public function delete( WC_Product $product, string $key ) {
 	public function get( WC_Product $product, string $key ) {
 		self::validate_meta_key( $key );
 
-		$value = get_post_meta( $product->get_id(), $this->prefix_meta_key( $key ), true );
+		$value = null;
+		if ( $product->meta_exists( $this->prefix_meta_key( $key ) ) ) {
+			$value = $product->get_meta( $this->prefix_meta_key( $key ), true );
 
-		if ( isset( self::TYPES[ $key ] ) && in_array( self::TYPES[ $key ], [ 'bool', 'boolean' ], true ) ) {
-			$value = wc_string_to_bool( $value );
+			if ( isset( self::TYPES[ $key ] ) && in_array( self::TYPES[ $key ], [ 'bool', 'boolean' ], true ) ) {
+				$value = wc_string_to_bool( $value );
+			}
```

Comment:
The code looks fine here, but we only have boolean attributes. In the attributes class I see a call to `wc_bool_to_string` but it doesn't have a matching `wc_string_to_bool`, is this part intended to be part of the attributes get function?

Reply:
ah yeah, that's intentional because we're using value objects for attributes and each object sets its own value type by overriding the `AttributeInterface::get_value_type` method and can implement a casting method by overriding `AbstractAttribute::cast_value`. 

The `cast_value` method in the parent `AbstractAttribute` class handles the basic casting based on the value returned by `get_value_type`. This method calls the `wc_string_to_bool` for all boolean attributes.